crates/driver/src/domain/mempools.rs

```diff
@@ -126,6 +117,17 @@ impl Mempools {
                     TxStatus::Executed => return Ok(hash.clone()),
                     TxStatus::Reverted => return Err(Error::Revert(hash.clone())),
                     TxStatus::Pending => {
+                        // Check if the current block reached the submission deadline block number
+                        if block.number == submission_deadline {
```

Comment:
In arbitrum we may not see every block (and in the spirit of defensive programming), let's use

```suggestion
                        if block.number >= submission_deadline {
```